### PR TITLE
Add deprecated notice.

### PR DESCRIPTION
**Pull request by tonycosentini**
This isn't really maintained anymore, probably best to point people to NullAway.

### CODE REVIEW

**Comment by tonycosentini** [comment]
cc @clonetwin26 (looks like I can't add you as a reviewer because you aren't in the Uber org)

File: README.md [hunk 1]
```diff
@@ -4,6 +4,10 @@
 
 This Gradle plug-in creates tasks to run [Infer](http://fbinfer.com) on Android and Java projects.
 
+## DEPRECATED
+
+This plug-in isn't used or maintained anymore. We strongly recomend migrating [NullAway](http://github.com/uber/nullaay) which is much more reliabile, significantly faster, and actively maintained.
```
Comment (msridhar) [inline README.md]: Can we just say that for null checking we recommend migrating to NullAway, without the other stuff? Infer checks other things, so NullAway doesn't subsume it. And also Infer is well maintained. I'd rather stick with something short and sweet ☺️

**Review by msridhar — CHANGES_REQUESTED** [review]

**Comment by ben-manes** [comment]
You may want to mention the Gradle versions this plugin is compatible with. Since NullAway had too many false positives, I tried this but it is incompatible with Gradle 4.4

```
* What went wrong:
A problem was found with the configuration of task ':caffeine:inferPrepare'.
> No value has been specified for property 'bootClasspath'.
...
Caused by: org.gradle.api.InvalidUserDataException: No value has been specified for property 'bootClasspath'.
        at org.gradle.api.internal.tasks.TaskValidationContext$Severity$2.report(TaskValidationContext.java:51)
```

File: README.md (hunk 1 above)
Comment (tonycosentini) [reply README.md]: Ooof, I meant that this plug-in is not currently maintained (and I think it works well for smaller projects, but I believe it has some pretty serious perf issues), thanks for catching that.

**Review by msridhar — APPROVED** [review]

**Pull request merged**